srcjs/input_rate.js

```diff
@@ -211,84 +213,109 @@ var InputBatchSender = function(shinyapp) {
   };
 }).call(InputBatchSender.prototype);
 
+
 var InputNoResendDecorator = function(target, initialValues) {
   this.target = target;
   this.lastSentValues = initialValues || {};
 };
 (function() {
-  this.setInput = function(name, value) {
+  this.setInput = function(name, value, opts) {
     const { name: inputName, inputType: inputType } = splitInputNameType(name);
     const jsonValue = JSON.stringify(value);
 
-    // Resend if either json value or the input type has changed.
+    // Resend if either json value or the input type has changed. Note that if
+    // `opts` changes, it will not the value to be resent.
     if (this.lastSentValues[inputName] &&
         this.lastSentValues[inputName].jsonValue === jsonValue &&
         this.lastSentValues[inputName].inputType === inputType) {
       return;
     }
     this.lastSentValues[inputName] = { jsonValue, inputType };
-    this.target.setInput(name, value);
+    this.target.setInput(name, value, opts);
   };
   this.reset = function(values) {
     values = values || {};
-    var strValues = {};
-    $.each(values, function(key, value) {
-      strValues[key] = JSON.stringify(value);
-    });
+    var strValues = mapValues(values, x => JSON.stringify(x.value));
     this.lastSentValues = strValues;
   };
 }).call(InputNoResendDecorator.prototype);
 
+
 var InputDeferDecorator = function(target) {
   this.target = target;
   this.pendingInput = {};
 };
 (function() {
-  this.setInput = function(name, value) {
+  this.setInput = function(name, value, opts) {
+    opts = Object.assign({
+      binding: null,
+      el: null
+    }, opts);
+
     if (/^\./.test(name))
-      this.target.setInput(name, value);
+      this.target.setInput(name, value, opts);
     else
-      this.pendingInput[name] = value;
+      this.pendingInput[name] = { value, opts };
   };
   this.submit = function() {
     for (var name in this.pendingInput) {
-      if (this.pendingInput.hasOwnProperty(name))
-        this.target.setInput(name, this.pendingInput[name]);
+      if (this.pendingInput.hasOwnProperty(name)) {
+        let input = this.pendingInput[name];
+        this.target.setInput(name, input.value, input.opts);
+      }
     }
   };
 }).call(InputDeferDecorator.prototype);
 
+
 var InputEventDecorator = function(target) {
   this.target = target;
 };
 (function() {
-  this.setInput = function(name, value, immediate) {
+  this.setInput = function(name, value, opts) {
+    opts = Object.assign({
```

Comment:
It might be better to have a global default options object instead of creating it each time.